src/commands/run.rs

```diff
@@ -337,36 +312,36 @@ pub async fn run(args: Args) -> Result<()> {
                     path = document.path()
                 ),
             }
-        } else if let Some(workflow) = document.workflow() {
-            (
-                workflow.name().to_owned(),
-                EngineInputs::Workflow(Default::default()),
-                origins,
-            )
         } else {
-            let mut tasks = document.tasks();
-            let first = tasks.next();
-            if tasks.next().is_some() {
-                bail!(
-                    "document `{path}` contains more than one task: use the `--name` option to \
-                     refer to a specific task by name",
-                    path = document.path()
-                )
-            } else if let Some(task) = first {
-                (
-                    task.name().to_owned(),
-                    EngineInputs::Task(Default::default()),
-                    origins,
-                )
-            } else {
-                bail!(
-                    "document `{path}` contains no workflow or task",
-                    path = document.path()
-                );
-            }
+            bail!("the `--entrypoint` option is required if no inputs are provided")
         }
     };
 
+    let output_dir = args
+        .output
+        .as_deref()
+        .unwrap_or_else(|| Path::new(&name))
+        .to_owned();
+
+    // Check to see if the output directory already exists and if it should be
+    // removed.
+    if output_dir.exists() {
+        if !args.overwrite {
+            bail!(
+                "output directory `{dir}` exists; use the `--overwrite` option to overwrite its \
+                 contents",
+                dir = output_dir.display()
+            );
+        }
+
+        std::fs::remove_dir_all(&output_dir).with_context(|| {
+            format!(
+                "failed to remove output directory `{dir}`",
+                dir = output_dir.display()
+            )
+        })?;
+    }
+
```

Comment:
I think it'll be confusing to have this fix for #143 land in the same commit as the behavior change for #141. Would you mind splitting out the #143 fix into a distinct PR so they won't get squashed together?

Reply:
This has been reverted. The change is trivial, but would have merge conflicts if put up right now due to the change from `name` to `entrypoint`, so I'll hold off till this merges and then put up a patch PR